src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1030,6 +1030,7 @@ public function withCount($relations)
                 $relation->getRelated()->newQuery(), $this
             );
 
+            $this->mergeModelDefinedRelationWheresToHasQuery($query, $relation);
             call_user_func($constraints, $query);
```

Comment:
Okay so I changed both, although I'm not really sure how to write an actual test for that specific case..